sdk/core/core/src/az_span.c

```diff
@@ -46,7 +46,13 @@ AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high
 
 AZ_NODISCARD AZ_INLINE uint8_t az_ascii_lower(uint8_t value)
```

Comment:
It looks like the function is internal, local to this file. I think we should name it as `_az_tolower()` i.e. starting with underscore.But without the underscore after `to`: we do `to_something` in other places, but here we would make a reference to the stdlib function `tolower()` which does the similar job. And given that we are referencing `tolower()`, I don;t think `_ascii_` is needed in the name.